### DISCUSSION TITLE
Setting the resolution for an extracted YouTube video

### DISCUSSION THREAD

**Discussion by prafulravindra** [omansak/libvideo#222] [Q&A]
Hi. I need help to download video with better resolution 720p/1080p including audio.
I have used 'GetVideo' in the code but the downloaded video is defaulted with '360p' resolution.

So I am trying out this in the code (max resolution is set as '1080p').
var videoInfos = youTube.GetAllVideos(link);
var maxResolution = videoInfos.FirstOrDefault(v => v.Resolution == 1080);

But how do I use 'GetAllVideos' to save the video with the resolution that I have set it in the code.

![image](https://user-images.githubusercontent.com/17843849/145000615-af03fbb1-1531-4219-a576-5263b20b5470.png)

**Comment by omansak** [comment]
Thank you,

```
var youTube = YouTube.Default; // starting point for YouTube actions
var videoInfos = youTube.GetAllVideosAsync(link).GetAwaiter().GetResult();
var maxResolution = videoInfos.First(i => i.Resolution == videoInfos.Max(j => j.Resolution));
```

You can look https://github.com/omansak/libvideo/blob/master/docs/README.md
Sample project
